src/methods.rs

```diff
@@ -92,6 +98,66 @@ impl LateLintPass for MethodsPass {
                     }
                 }
             }
+            // check Option.map(_).unwrap_or(_)
+            else if name.node.as_str() == "unwrap_or" {
+                if let ExprMethodCall(ref inner_name, _, ref inner_args) = args[0].node {
+                    if inner_name.node.as_str() == "map"
+                            && match_type(cx, cx.tcx.expr_ty(&inner_args[0]), &OPTION_PATH) {
+                        // lint message
+                        let msg =
+                            "called `map(f).unwrap_or(a)` on an Option value. This can be done \
+                             more directly by calling `map_or(a, f)` instead";
+                        // get args to map() and unwrap_or()
+                        let map_arg = snippet(cx, inner_args[1].span, "..");
+                        let unwrap_arg = snippet(cx, args[1].span, "..");
+                        // lint, with note if neither arg is > 1 line and both map() and
+                        // unwrap_or() have the same span
+                        let multiline = map_arg.lines().count() > 1
+                                        || unwrap_arg.lines().count() > 1;
+                        let same_span = inner_args[1].span.expn_id == args[1].span.expn_id;
+                        if same_span && !multiline {
+                            span_note_and_lint(
+                                cx, OPTION_MAP_UNWRAP_OR, expr.span, msg, inner_name.span,
```

Comment:
Shouldn't this be `expr.span` both times? We're asking them to replace the entire expr.